lib/checkautovariables.cpp

```diff
@@ -648,6 +648,7 @@ void CheckAutoVariables::checkVarLifetimeScope(const Token * start, const Token
                     if (var && !var->isLocal() && !var->isArgument() && !(val.tokvalue && val.tokvalue->variable() && val.tokvalue->variable()->isStatic()) &&
                         !isVariableChanged(nextTok,
                                            tok->scope()->bodyEnd,
+                                           var->valueType() ? var->valueType()->pointer : 0,
```

Comment:
As I recall the pointer value is different than the indirect value.

Reply:
Then how do we get the correct value here?

Reply:
Is the pointer a bitset that sets each level indirection? Or is that just const?

Reply:
`///< 0=>not pointer, 1=>*, 2=>**, 3=>***, etc`
`constness` and `volatileness` are bitsets.

Reply:
Sorry I was confused.